packages/jseql/src/identify/index.ts

```diff
@@ -0,0 +1,83 @@
+export type CtsRegions = 'ap-northeast-1' | 'us-east-1' | 'us-west-2' | 'eu-west-1' |
+'ap-southeast-1' | 'us-west-1' | 'eu-central-1' | 'eu-west-2' | 'ap-south-1' |
+'us-east-2' | 'ap-northeast-2' | 'ca-central-1' | 'ap-southeast-2' | 'eu-north-1' |
+'sa-east-1' | 'eu-west-3' | 'ap-east-1' | 'me-south-1' | 'af-south-1'
+
+export type Context = {
+  identityClaim: string[]
+  workspaceId: string
+  region: CtsRegions
+}
+
+export type IdentifyOptions = {
+  fetchFromCts?: boolean
+}
+
+export class LockContext {
+  private cts_token: string | undefined
+  private workspaceId: string
+  private region: CtsRegions = 'ap-southeast-2'
+  private context: Context
+
+  constructor(context: Context) {
+    this.region = context.region ?? this.region
+    this.workspaceId = context.workspaceId
+    this.context = context
+  }
+
+  async identify(jwtToken: string, {
+    fetchFromCts = true,
+  }: IdentifyOptions = {}): Promise<LockContext> {
+    const workspaceId = this.workspaceId
+    const region = this.region
+
+    if (fetchFromCts) {
+      const ctsResponse = await fetch(
+        `https://${region}.aws.auth.viturhosted.net/api/federate`,
```

Comment:
We currently only support ap-southeast-2 for the region here. ZeroKMS supports seven different regions, but CTS currently only runs in ap-southeast-2.

Reply:
I think we'll also want to push more of the logic here down into cipherstash-client/jseql-ffi. Probably OK to handle in TS in the short term, but long term we'll want to share this logic with Proxy and other language clients.

I can take a look at moving this into cipherstash-client/jseql-ffi after the CTS changes land.